### PR TITLE
Feature demo user #18

### LINKED ISSUE #18: FE: Demo User
This ticket does not have a mock-up so please add in the button in the Login and Sign up pages and make sure the design is consistent with the overall look and feel.

- Create a "demo user" functionality that will be used later to allow recruiters and managers to easily test the app's features without having to sign up
- Add a button to allow for signing in as a demo user
- This is effectively a login button with a hardcoded username and password
- Show button both on signup and login pages
- In production you should manually create a user and some test data that would best show off your project. Do this in development right now to test this feature

### PR DESCRIPTION
**Pull request by MarianaES**
### What this PR does (required):
- Add the option to sign up/log in as a demo Owner or Sitter user using hardcoded data (username, email and password). 

### Screenshots / Videos (front-end only):
Sign up view
https://user-images.githubusercontent.com/29843870/139518193-7ea6d7ef-a079-4271-a92b-ee422e6e3442.mov

Log in view
https://user-images.githubusercontent.com/29843870/139518199-b6b8346c-3b29-4a58-929b-6c8c9259d613.mov

### Any information needed to test this feature (required):
- Sing up or Log in clicking on the owner or sitter options for demo.

### CODE REVIEW

**Review by sundayezeilo — COMMENTED** [review]
This one is good!

File: client/src/pages/SignUp/SignUpForm/SignUpForm.tsx [hunk 1]
```diff
@@ -33,12 +35,19 @@ interface Props {
 const SignUpForm = ({ handleSubmit }: Props): JSX.Element => {
   const classes = useStyles();
 
+  const [demoUser, setDemoUser] = useState<Record<string, string> | undefined>();
```
Comment (rajivtitus) [inline client/src/pages/SignUp/SignUpForm/SignUpForm.tsx:38]: We don't a state to store the demo user's info - we can simplify the implementation by storing the demo user in an object :)

File: client/src/components/DemoSelect/DemoSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    user({ username, email, password });
+  }
+
+  return (
+    <>
+      <Divider variant="middle" />
+      <Typography className={classes.header} align="center" color="textPrimary">
+        {label}
+      </Typography>
+      <Chip label="Owner" icon={<PetsIcon />} onClick={() => setUserDemo(mockOwnerUser)} className={classes.demo} />
```
Comment (rajivtitus) [inline client/src/components/DemoSelect/DemoSelect.tsx:42]: As opposed to setting the state in the parent component, why don't we call the login function here itself? 😄

**Review by rajivtitus — COMMENTED** [review]
Good work here! Just a couple of comments for you to have a look :)

File: client/src/components/DemoSelect/DemoSelect.tsx (hunk 2 above)
Comment (MarianaES) [reply client/src/components/DemoSelect/DemoSelect.tsx:42]: As the code is just now, my demo button is just populating the forms for Log In and Sign Up so the user can click the Login and Create button respectively after that.  Those buttons keep maintaining the responsibility of the login.   Maybe it's over complicated 😅 I'll change it.

**Review by rajivtitus — APPROVED** [review]
As discussed in our meeting, we can leave the implementation as is to focus on other features 😄

**Pull request merged**